jans-linux-setup/jans_setup/setup_app/utils/package_utils.py

```
@@ -69,20 +69,22 @@ def check_and_install_packages(self):
-                            base.download(f'https://downloads.mysql.com/archives/get/p/23/file/{deb_fn}', os.path.join(tmpdirname, deb_fn))
-                        self.run([install_command.format(os.path.join(tmpdirname, '*.deb'))],  shell=True)
+                        apt_deb_fn = 'mysql-apt-config_0.8.39-1_all.deb'
+                        local_apt_path = os.path.join(tmpdirname, apt_deb_fn)
+                        base.download(f'https://repo.mysql.com//{apt_deb_fn}', local_apt_path)
+                        if not os.path.exists(local_apt_path):
+                            print(f"Failed to download {apt_deb_fn}, can't continue...")
+                            sys.exit(1)
+                        self.run([install_command.format(local_apt_path)], shell=True)
+                        self.run(update_command, shell=True, get_stderr=True)
+                        # check if MySQL Server is installable via the newly configured apt repo
+                        policy_out = self.run("apt-cache policy mysql-server", shell=True)
```

Ruff (0.15.11)

[error] Function call with `shell=True` parameter identified, security issue

(S604)